File: 2824-count-pairs-whose-sum-is-less-than-target/2824-count-pairs-whose-sum-is-less-than-target.py
class Solution:
    def countPairs(self, nums: List[int], target: int) -> int:
        nums.sort()
        left, right, output = 0,len(nums)-1,0
        
        while left < right:
            if nums[left]+nums[right] < target:
                output+=(right-left)
                left+=1
            else:
                while(right>=0 and nums[left]+nums[right] >= target):
                    right-=1
        return output

File: 2824-count-pairs-whose-sum-is-less-than-target/2824-count-pairs-whose-sum-is-less-than-target.java
class Solution {
    public int countPairs(List<Integer> nums, int target) {
        Collections.sort(nums);
        int left = 0;
        int right = nums.size() - 1;
        int output = 0;
        
        while(left < right){
            if(nums.get(left)+nums.get(right) < target){
                output+= (right-left);
                left+=1;
            }
            else{
                while(right >=0 && nums.get(left)+nums.get(right) >= target){
                    right-=1;
                }
            }
        }
        
        return output;
        
    }
}